ext/node/polyfill.rs

```diff
@@ -141,6 +141,10 @@ pub static SUPPORTED_BUILTIN_NODE_MODULES: &[NodeModulePolyfill] = &[
     name: "process",
     specifier: "ext:deno_node/process.ts",
   },
+  NodeModulePolyfill {
+    name: "punycode",
+    specifier: "ext:deno_node/punycode.ts",
+  },
```

Comment:
> The version of the punycode module bundled in Node.js is being deprecated. In a future major version of Node.js this module will be removed. Users currently depending on the punycode module should switch to using the userland-provided [Punycode.js](https://github.com/bestiejs/punycode.js) module instead.

Should we bother adding this if it's going to be deprecated?

Reply:
I think so, currently existing packages might still depend on it and it doesn't say when it will be removed. @kt3k thoughts?

Reply:
(and we already support it in CJS just not ESM)

Reply:
I'm fine with either way, but probably it's less confusing if we already provide it via cjs. I confirmed the below prints the punycode module without error:

```js
import { createRequire } from "node:module";
console.log(createRequire(import.meta.url)("node:punycode"));
```